sql-plugin/src/main/scala/com/nvidia/spark/rapids/io/async/AsyncOutputStream.scala

```diff
@@ -112,7 +121,9 @@ class AsyncOutputStream(openFn: Callable[OutputStream], trafficController: Traff
   }
 
   override def write(b: Array[Byte]): Unit = {
-    scheduleWrite(() => delegate.write(b), b.length)
+    val copy = new Array[Byte](b.length)
```

Comment:
the allocation here is probably going to be something we want to do differently in the future. I don't think we have to do it in this PR. I just think we need to bench it and likely go towards a circular buffer or something like that. The circular buffer could serve as a throttle, blocking `write`.

Reply:
It would be interesting to see the bench result.